src/main/java/com/selfxdsd/selfpm/ReviewContractsMarkedForRemoval.java

```diff
@@ -0,0 +1,204 @@
+ */
+@Component
+public final class ReviewContractsMarkedForRemoval {
+
+    /**
+     * The PMs will review contracts every day.
+     */
+    private static final long EVERY_24_HOURS = 86_400_000L;
+
+
+    /**
+     * The PMs will review contracts every day with a start delay of 15 minutes.
+     */
+    private static final long DELAY_15_MINUTES = 900_000L;
+
+    /**
+     * Logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(
+        ReviewContractsMarkedForRemoval.class
+    );
+
+    /**
+     * Self's core.
+     */
+    private final Self selfCore;
+
+    /**
+     * Time "now" supplier that will be used to determined how many days
+     * ago a Contract marked for removal.
+     */
+    private final Supplier<LocalDateTime> nowSupplier;
+
+    /**
+     * Ctor.
+     * @param selfCore Self Core, injected by Spring automatically.
+     */
+    @Autowired
+    public ReviewContractsMarkedForRemoval(final Self selfCore) {
+        this(selfCore, LocalDateTime::now);
+    }
+
+    /**
+     * Ctor used in tests.
+     * @param selfCore Self Core.
+     * @param nowSupplier Time "now" supplier.
+     */
+    ReviewContractsMarkedForRemoval(final Self selfCore,
+                                    final Supplier<LocalDateTime> nowSupplier) {
+        this.selfCore = selfCore;
+        this.nowSupplier = nowSupplier;
+    }
+
+    /**
+     * Every 24 hours the PMs should verify contracts marked for removal.
+     * <br/>
+     * Contracts marked for removal more than 30 days ago will be removed.
+     * <br/>
+     * It also has start of 15 minutes delay, so it will not overlap with
+     * other scheduled jobs.
+     * @checkstyle NestedForDepth (50 lines).
+     */
+    @Scheduled(fixedRate = EVERY_24_HOURS, initialDelay = DELAY_15_MINUTES)
+    public void reviewContractsMarkedForRemoval() {
+        LOG.debug("PMs reviewing project contracts marked for removal...");
+        for(final ProjectManager manager : this.selfCore.projectManagers()) {
+            LOG.debug(
+                "PM @" + manager.username()
+                + " reviewing their project contracts marked for removal..."
+            );
+            for(final Project project : manager.projects()) {
+                LOG.debug(
+                    "Reviewing contracts marked for removal from project "
+                    + project.repoFullName() + " at " + project.provider()
+                    + "... "
+                );
+                final List<Contract> toRemove = this.contractsToRemove(project);
+                LOG.debug(
+                    "For project " + project.repoFullName() + " at "
+                        + project.provider() + " there are " + toRemove.size()
+                        + " contracts that will be removed..."
+                );
+                for(final Contract contract: toRemove){
+                    project.resolve(this.contractRemovalEvent(contract));
```

Comment:
@amihaiemil I've used here the Event API to verify in tests which contracts are removed,
until we have an API for that.

Reply:
@amihaiemil On a 2nd thought, using the events api in a way that wasn't intented is a bad idea, luckly `StoredProjec`t dosen't throw if the event is unknown. Just injecting a simple callback in testing constructor is much more clearer aproach.